packages/fela-bindings/src/FelaComponentFactory.js

```diff
@@ -1,18 +1,48 @@
 /* @flow */
+//import assignStyle from 'css-in-js-utils/lib/assignStyle'
+import arrayReduce from 'fast-loops/lib/arrayReduce'
+
+function assignStyle(a, b) {
+  return Object.assign({}, a, b)
+}
+function combineRules(...rules: Array<Function>): Function {
+  return (props, renderer) =>
+    arrayReduce(
+      rules,
+      (style, rule) =>
+        assignStyle(
+          style,
+          typeof rule === 'function' ? rule(props, renderer) : rule
+        ),
+      {}
+    )
+}
```

Comment:
Wouldn't this potentially clobber media queries?

Consider:
```jsx
const style = props => ({
  '@media (min-width: 600px)': { color: 'red', }
})

const MyComponent = props => (<FelaComponent style={style}  {...props}/>)

cont extendedStyle = props => ({
  '@media (min-width: 600px)': { backgroundColor: '#000', }
})

const MyExtendedComponent = props (<MyComponent extend={extendedStyle} {...props}
```

If I'm not mistaken, because `assignStyle` uses `Object.assign`, it only provides shallow merging of the resulting styles, and the media query in `extendedStyle` will not extend the one in `style`, but clobber it. From a user perspective, I'd expect an `extend` attribute to act in the same way as `fela-plugin-extend`.

Reply:
Oh sure, should have added a comment. This Object.assign was only used to test the approach. I'll be using assignStyle which allows deep merging of objects and even merges arrays correctly with the next version (which also makes fela-combine-arrays obsolete and reduces complexity again).

Reply:
I think we should also consider making `fela-plugin-extend` part of Fela's core, as to provide a consistent API throughout. Since `assignStyle` will be used by `FelaComponent` by default, it won't even add anything to the bundle size

Reply:
but the extend prop works different than the extend plugin.
The extend plugin is more like a nice API to do conditional style extensions. Not sure about that addition.
As soon as you add the plugin, you can of course also do sth. like:

```js
const extend = ({ variant }) => ({
  fontSize: 20,
  extend: [{
    condition: variant === 'primary',
    style: {
      color: 'red',
      backgroundColor: 'lightred'
    }
  },{
    condition: variant === 'secondary',
    style: {
      color: 'blue',
      backgroundColor: 'lightblue'
    }
  }]
})

const ExtendedButton = () => <Button extend={style} />
```

Reply:
But extend is not only for conditional extension, it is also for things like this:

```js
const fromDesktop = (prop, value) => ({ '@media (min-width: 1024px)': { [prop]: value, }, })

const style = props => ({
  extend:[
    fromDesktop('color', 'red'),
    fromDesktop('overflow', 'hidden'),
  ],
})
```